CMakeLists.txt

```diff
@@ -58,6 +63,51 @@ find_package(Doxygen)
 if(Doxygen_FOUND)
   doxygen_add_docs(doxygen core)
 endif()
+
+#------------------------------------------------------------------------------#
+# Architecture
+#------------------------------------------------------------------------------#
+if(BUILD_MARCH AND BUILD_MCPU)
+  message(FATAL_ERROR "BUILD_MARCH and BUILD_MCPU are incompatible")
+endif()
+
+# Try -march first. On platforms that don't support it, GCC will issue
+# a hard error, so we'll know not to use it.
+if(BUILD_MARCH)
```

Comment:
is `BUILD_MARCH` a native CMake variable?

Reply:
No, we still need to a cache doc string for it!

Reply:
ah OK - didn't see it elsewhere in the configure